server/internal/okta/client.go

```diff
@@ -0,0 +1,107 @@
+package okta
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/llmariner/rbac-manager/server/internal/token"
+)
+
+var _ token.Client = &defaultClient{}
+
+// NewDefaultClient returns a new default client.
+func NewDefaultClient() token.Client {
+	return &defaultClient{}
+}
+
+type defaultClient struct{}
+
+// TokenIntrospect introspects the given token.
+func (c *defaultClient) TokenIntrospect(tokenStr string) (*token.Introspection, error) {
+	if !strings.HasPrefix(tokenStr, "Bearer ") {
+		return nil, fmt.Errorf("unexpected form of auth header")
+	}
+	accessToken := tokenStr[7:]
+	claims, err := c.getClaimsFromAccessToken(accessToken)
+	if err != nil {
+		return nil, fmt.Errorf("unexpected form of claims: %s", err)
+	}
+
+	email, err := c.getEmail(claims)
+	if err != nil {
+		return nil, fmt.Errorf("could not get email claim: %s", err)
+	}
+
+	userID, err := c.getUserID(claims)
+	if err != nil {
+		return nil, fmt.Errorf("could not get user ID: %s", err)
+	}
+	fmt.Printf("Found email[%s], userID[%s] from claims %+v\n", email, userID, claims)
+
+	return &token.Introspection{
+		Active:  true,
+		Subject: userID,
+		Extra: token.IntrospectionExtra{
+			Email: email,
+		},
+	}, nil
+}
+
+// getClaimsFromAccessToken gets the claims from the JWT access token.
+func (c *defaultClient) getClaimsFromAccessToken(accessToken string) (jwt.MapClaims, error) {
+	// Decode the JWT token. Pass nil to keyFunc to skip
+	// validation. The access token should have already been
```

Comment:
This is currently not true for LLMarienr backend as we don't configure a Kong plugin in the ingress controller (like below). 

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    konghq.com/override: customization
    konghq.com/plugins: jwt-cluster
```

We could enable if all requests should come with JWT issued by Okta instead of Dex. Let me handle as we cannot switch simply. We first need to fix the frontend.